lib/ansible/module_utils/urls.py

```diff
@@ -1723,14 +1723,20 @@ def prepare_multipart(fields):
                 except Exception:
                     mime = 'application/octet-stream'
             main_type, sep, sub_type = mime.partition('/')
+
+            multipart_encoding_str = value.get("multipart_encoding")
+            if not multipart_encoding_str:
+                multipart_encoding_str = "base64"
         else:
             raise TypeError(
                 'value must be a string, or mapping, cannot be type %s' % value.__class__.__name__
             )
 
+        multipart_encoding = set_multipart_encoding(multipart_encoding_str)
```

Comment:
Move this into the block below (`if not content and filename:`) since it's only applicable in that scope.